opm/parser/eclipse/EclipseState/Summary/Summary.cpp

```diff
@@ -79,30 +82,114 @@ namespace Opm {
+                                               int init = 0 )
+    {
+        return {
+            record.getItem( init + 0 ).get< int >( 0 ) - 1,
+            record.getItem( init + 1 ).get< int >( 0 ) - 1,
+            record.getItem( init + 2 ).get< int >( 0 ) - 1
+        };
+    }
+
+    static inline std::array< int, 3 > getijk( const Completion& completion ) {
+        return { completion.getI(), completion.getJ(), completion.getK() };
+    }
+
     static inline std::vector< ERT::smspec_node > keywordB(
             const DeckKeyword& keyword,
             const EclipseState& es ) {
 
-        std::array< int, 3 > dims = {{
-            int( es.getEclipseGrid()->getNX() ),
-            int( es.getEclipseGrid()->getNY() ),
-            int( es.getEclipseGrid()->getNZ() )
-        }};
+        auto dims = dimensions( *es.getEclipseGrid() );
 
         const auto mkrecord = [&dims,&keyword]( const DeckRecord& record ) {
-
-            std::array< int , 3 > ijk = {{
-                record.getItem( 0 ).get< int >( 0 ) - 1,
-                record.getItem( 1 ).get< int >( 0 ) - 1,
-                record.getItem( 2 ).get< int >( 0 ) - 1
-            }};
-
+            auto ijk = getijk( record );
             return ERT::smspec_node( keyword.name(), dims.data(), ijk.data() );
         };
 
         return fun::map( mkrecord, keyword );
     }
 
+    static inline std::vector< ERT::smspec_node > keywordR(
+            const DeckKeyword& keyword,
+            const EclipseState& es ) {
+
+        auto dims = dimensions( *es.getEclipseGrid() );
+
+        const auto mknode = [&dims,&keyword]( int region ) {
+            return ERT::smspec_node( keyword.name(), dims.data(), region );
+        };
+
+        const auto& item = keyword.getDataRecord().getDataItem();
+        const auto regions = item.size() > 0 && item.hasValue( 0 )
+            ? item.getData< int >()
+            : es.getRegions();
+
+        return fun::map( mknode, regions );
+    }
+
+   static inline std::vector< ERT::smspec_node > keywordC(
+           const DeckKeyword& keyword,
+           const EclipseState& es ) {
+
+       std::vector< ERT::smspec_node > nodes;
+       const auto& keywordstring = keyword.name();
+       const auto& schedule = es.getSchedule();
+       const auto last_timestep = schedule->getTimeMap()->last();
+       auto dims = dimensions( *es.getEclipseGrid() );
+
+       for( const auto& record : keyword ) {
+
+           const auto& wellname = record.getItem( 0 ).get< std::string >( 0 );
+
+           if( wellname == "*" ) {
+               for( const auto& well : schedule->getWells() ) {
+
+                   const auto& name = wellName( well );
+
+                   for( const auto& completion : *well->getCompletions( last_timestep ) ) {
+                       auto cijk = getijk( *completion );
+
+                       /* well defaulted, block coordinates defaulted */
+                       if( record.size() != 4 ) {
```

Comment:
This test will never kick in - the `DeckRecord` will *always* have size 4, but it might consist of defaulted items. So the test should rather be based on checking the `defaultApplied()` of one of the coordinate items.